Assets/MixedRealityToolkit/Inspectors/ChannelPackerWindow.cs

```diff
@@ -0,0 +1,238 @@
+                                                                                                     (Texture2D)standardMaterial.GetTexture("_MainTex");
+                smoothnessMapChannel = smoothnessMap != null ? Channel.Alpha : smoothnessMapChannel;
+            }
+        }
+
+        private void Save()
+        {
+            int width;
+            int height;
+            Texture[] textures = new Texture[] { metallicMap, occlusionMap, emissionMap, smoothnessMap };
+            CalculateChannelMapSize(textures, out width, out height);
+
+            Texture2D channelMap = new Texture2D(width, height);
+            RenderTexture renderTexture = RenderTexture.GetTemporary(width, height, 0, RenderTextureFormat.Default, RenderTextureReadWrite.Linear);
+
+            // Use the GPU to pack the various texture maps into a single texture.
+            Material channelPacker = new Material(Shader.Find("Hidden/ChannelPacker"));
+            channelPacker.SetTexture("_MetallicMap", metallicMap);
+            channelPacker.SetInt("_MetallicMapChannel", (int)metallicMapChannel);
+            channelPacker.SetFloat("_MetallicUniform", metallicUniform);
+            channelPacker.SetTexture("_OcclusionMap", occlusionMap);
+            channelPacker.SetInt("_OcclusionMapChannel", (int)occlusionMapChannel);
+            channelPacker.SetFloat("_OcclusionUniform", occlusionUniform);
+            channelPacker.SetTexture("_EmissionMap", emissionMap);
+            channelPacker.SetInt("_EmissionMapChannel", (int)emissionMapChannel);
+            channelPacker.SetFloat("_EmissionUniform", emissionUniform);
+            channelPacker.SetTexture("_SmoothnessMap", smoothnessMap);
+            channelPacker.SetInt("_SmoothnessMapChannel", (int)smoothnessMapChannel);
+            channelPacker.SetFloat("_SmoothnessUniform", smoothnessUniform);
+            Graphics.Blit(null, renderTexture, channelPacker);
+            DestroyImmediate(channelPacker);
+
+            // Save the last render texture to a texture.
+            RenderTexture previous = RenderTexture.active;
+            RenderTexture.active = renderTexture;
+            channelMap.ReadPixels(new Rect(0.0f, 0.0f, width, height), 0, 0);
+            channelMap.Apply();
+            RenderTexture.active = previous;
+            RenderTexture.ReleaseTemporary(renderTexture);
+
+            // Save the texture to disk.
+            string filename = string.Format("{0}{1}.png", GetChannelMapName(textures), "_Channel");
+            string path = EditorUtility.SaveFilePanel("Save Channel Map", "", filename, "png");
+
+            if (path.Length != 0)
+            {
+                byte[] pngData = channelMap.EncodeToPNG();
+
+                if (pngData != null)
+                {
+                    File.WriteAllBytes(path, pngData);
+                    Debug.LogFormat("Saved channel map to: {0}", path);
```

Comment:
Should this notification maybe be a dialog box with an option to open the folder that the PNG is located in?

Reply:
Likely better to specify a save path in the window, not in a dialog.

Reply:
This is a great usability idea! I'm on the fence if this would be useful to the user or an annoyance to users who are processing tons of textures to have to hit "ok" after every save?

Reply:
I'm also worried about the automated import process being interrupted with a dialog as well.

Reply:
A dialog would be kind of intrusive, I agree with you guys.